File: server/src/game/conf/ParticleConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type ParticleConfigElem struct {
	Id int
	Texture string
	Config string
	Location []int
	Scale []float32
	
}

type ParticleConfig0 struct {
	Dic map[int]*ParticleConfigElem
	Arr []*ParticleConfigElem
}

var (
	ParticleConfig ParticleConfig0
)

func loadParticleConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	ParticleConfig.Dic = make(map[int]*ParticleConfigElem)
	ParticleConfig.Arr = make([]*ParticleConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &ParticleConfig.Arr)
	for _,conf := range ParticleConfig.Arr{
		ParticleConfig.Dic[conf.Id] = conf
	}
	
}

File: client/src/game/conf/ParticleConfig.ts
class ParticleConfig {
    private static _dic:Dictionary<ParticleConfig> = new Dictionary<ParticleConfig>();
    private static _arr:Array<ParticleConfig> = new Array<ParticleConfig>();

	 public Id :number ;
     public Texture :string ;
     public Config :string ;
     public Location : Array<number>  =new Array<number>();
     public Scale : Array<number>  =new Array<number>();
     public SoundId :number ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<ParticleConfig>{
        return  ParticleConfig._dic;
    }
    public static get Arr():Array<ParticleConfig>{
        return ParticleConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:ParticleConfig = <ParticleConfig>json[i];
				ParticleConfig._arr.push(config);
				ParticleConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("ParticleConfig loader fail!");
        }
    }
}